Move to monorepo structure and add SDK cli, compiler and node-client

mslavov opened the pull request:
Created a Lerna monorepo
Moved the `api-clients` -> `packages/angular`
Added
`cli` - for working with vCD extensibility solutions
`compiler` - for building defined entity types
`node-client` - for working with the vCD API from NodeJS

Testing Done:
- Verified `angular` can be build successfully
- Verified `cli`, `compiler` and `node-client` build and work locally

Signed-off-by: Milko Slavov <mslavov@vmware.com>

nvladimirovi commented:
Hello @mslavov,

Can you please describe what changes you did in the commit message and the PR description + Testing Done section.

mslavov commented:
Created a Lerna monorepo
Moved the `api-clients` -> `packages/angular`
Added 
`cli` - for working with vCD extensibility solutions
`compiler` - for building defined entity types
`node-client` - for working with the vCD API from NodeJS

nvladimirovi commented on packages/cli/.editorconfig:
```diff
@@ -0,0 +1,11 @@
+root = true
```
Maybe we've to ignore this file? Is it related somehow with the build process?

mslavov replied:
I'm not sure where it comes from. Do you know?

mslavov replied:
Looking what it is: https://editorconfig.org it seems valuable and should reside into the repo, since it aims to make development consistent.

nvladimirovi commented on packages/cli/src/deployer/interfaces.ts:
```diff
@@ -0,0 +1,91 @@
+                return prev
+            }, {})
+        this.debug(existingInts)
+        return Promise.all(
```
Just a note. If you have huge list of files this could potentially throw OOM Exception.

mslavov replied:
This is a temporary implementation, since with the package support it will be done differently. On the other hand we don't need to support huge list of file I think, or at least not at this point.
